p2p/src/net/default_backend/mod.rs

```diff
@@ -203,10 +212,10 @@ impl<T: TransportSocket> MessagingService for MessagingHandle<T> {
 
     fn make_announcement(&mut self, announcement: Announcement) -> crate::Result<()> {
         let message = announcement.encode();
-        if message.len() > ANNOUNCEMENT_MAX_SIZE {
+        if message.len() > *self.p2p_config.max_message_size {
```

Comment:
I think there are some problems with this check:
- Checking the encoded `Announcement` size does not guarantee that the final message size will be less than `max_message_size` (very unlikely to be happen in practice).
- We don't check the message size in `send_message` where it is more likely to fail.
- We should never send such large announcements to trigger this.

Maybe it would be better to remove this check?

Reply:
I agree with your reasoning, so I have removed useless checks and added a few new tests.